### PR TITLE
Add BACK_TO_TOP event and fire when button is clicked

### LINKED ISSUE #2276: Analytics event: `BACK_TO_TOP`
## Description

The UI component assumes that users need to go back to the top part of Openverse interface, and this change would help us to understand whether that is true or if any other action derivates from this search flow.

Steps:
1. Add `BACK_TO_TOP` to the `Events` type: https://github.com/WordPress/openverse/blob/839418fabb0972ec3c3ccf48c72fd64cbba03cb9/frontend/src/types/analytics.ts#L12
    - Add `query` and `page` payload for the query term and requested page depth respectively. Additionally, `scrolledPixels` derived from `document.getElementById('main-page').scrollTop` (or, preferably a Vue ref on that element instead) and similarly `maxScroll` derived from `.scrollTopMax` off the same `#main-page` element.
2. Fire the event using the `useAnalytics` composable when the back to top button is clicked. It should be added to https://github.com//WordPress/openverse/blob/HEAD/frontend/src/components/VScrollButton.vue#L47-L50

## Additional context

Screenshot of the UI component mentioned

Screenshot of openverse's image results. 'Back to top' is highlighted 

## Additional information

- [RFC](https://docs.openverse.org/projects/proposals/analytics/20221006-implementation_plan_frontend.html#events)
- [Example usage in `VHomeGallery.vue`](https://github.com/WordPress/openverse/blob/839418fabb0972ec3c3ccf48c72fd64cbba03cb9/frontend/src/components/VHomeGallery/VHomeGallery.vue#L128-L134)

### LINKED ISSUE #4948: Button to return to top not working in search page
## Description
In the search results page, clicking the "return to top" button at the bottom-right of the page does not work.

## Reproduction
1. Run a search.
2. Click the `^` button at the bottom-right of the page.
3. The page does not scroll back to the top.

## Screenshots
Screenshot 2024-09-17 at 1 53 28 PM

## Environment
- Device: 15in M2 Macbook Air
- OS: MacOS Sonoma 14.6.1
- Browser: Firefox 130

## Additional context
The button still works on other pages, such the page for an individual piece of content, or the [Sources](https://openverse.org/sources) page.

### PR DESCRIPTION
**Pull request by d3jawu**
## Fixes
Resolves #2276 by @fcoveram.

Fixes #4948 by @d3jawu.

## Description
Adds a new analytics event, `BACK_TO_TOP`, and fires it when the "back to top" button in the bottom-right corner is clicked.

Fixes the behavior of the "back to top" button, which had stopped working on the search page. Verified that the behavior still works on other pages.

## Testing Instructions
1. Run a search.
2. Click the "back to top" caret button in the bottom-right corner.
3. See that an analytics event for `BACK_TO_TOP` is logged.
4. See that the page scrolls back to the top instead of remaining in place.

## Checklist
- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (`ov just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/components/VScrollButton.vue [hunk 1]
```diff
@@ -9,15 +13,28 @@ const props = withDefaults(
   { isFilterSidebarVisible: true }
 )
 
+const { sendCustomEvent } = useAnalytics()
+const searchStore = useSearchStore()
+const mediaStore = useMediaStore()
 defineEmits<{ tab: [KeyboardEvent] }>()
```
Comment (zackkrida) [inline frontend/src/components/VScrollButton.vue:19]:
```suggestion
const mediaStore = useMediaStore()

defineEmits<{ tab: [KeyboardEvent] }>()
```

File: frontend/src/components/VScrollButton.vue [hunk 2]
```diff
@@ -9,15 +13,28 @@ const props = withDefaults(
...
 
 const hClass = computed(() =>
   props.isFilterSidebarVisible ? positionWithSidebar : positionWithoutSidebar
 )
 const scrollToTop = (e: MouseEvent) => {
```
Comment (zackkrida) [inline frontend/src/components/VScrollButton.vue:25]:
```suggestion
const scrollToTop = (_: MouseEvent) => {
```

**Review by obulat — CHANGES_REQUESTED** [review]
This is a great start ✨ 

The scroll button is on all scrollable pages, including such content pages as https://openverse.org/about, so we should not send the event when we are not on a search or a collection page.

To check the page we are on, we can use `useRoute`, and check if the `route.name` is one of `search__image`, `search__audio`, `search`, `image__collection`, `audio_collection`. Note that the route names are usually `Symbol`s, so you would need to use `.toString()` on them, and then also remove the language suffix such as `__en`.

Also, this is not mentioned in the issue, but I think the params should be similar to the once sent by `REACH_RESULT_END`:
https://github.com/WordPress/openverse/blob/749259effdc9f6695ef13711b499d32f53dcd9ea/frontend/src/types/analytics.ts#L116

Please let me know if anything is unclear. If you think this is too much work, we can remove the events from this PR, and just ship the fix to scrolling on the search pages.

**Comment by d3jawu** [comment]
> I think the params should be similar to the once sent by REACH_RESULT_END:

Hi @obulat what exactly do you mean by this? Should they match them exactly, or include them in addition to the ones listed in the issue?

**Comment by d3jawu** [comment]
A couple little things:
- I've used -1 instead of `undefined` for the numeric values.
- For unsupported search types (`video` and `model-3d`) I've defaulted to `all`.

Let me know if you'd rather I use other values!

**Review by obulat — CHANGES_REQUESTED** [review]
This works for the search pages, but needs to be updated for other pages.
I have several suggestions:
1. I did not give the correct route names in the previous suggestions. They should be fixed.
2. The search pages scroll the main page, but all others scroll the body. This means that the code for `scrollPixels` and `maxScroll` works correctly only on the search pages.
3. Since the scroll values are calculated after the page was scrolled to top, they are set to 0 (or -1, because of the `|| -1` condition).
4. It's better to get the search type from the `mediaStore._searchType`, which filters out the unsupported types.

Here's a patch with suggestions on how to fix these problems:

```
Subject: [PATCH] Fix search type, route names and scroll values
---
Index: frontend/nuxt.config.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/frontend/nuxt.config.ts b/frontend/nuxt.config.ts
--- a/frontend/nuxt.config.ts	(revision 431ccc43aaa8f58cacf18170c8f5cbddde48cbbd)
+++ b/frontend/nuxt.config.ts	(date 1727112312513)
@@ -42,7 +42,7 @@
       },
       plausible: {
         ignoredHostnames: ["localhost", "staging.openverse.org"],
-        logIgnoredEvents: true,
+        logIgnoredEvents: false,
         apiHost: "http://localhost:50290",
         domain: "localhost",
       },
Index: frontend/src/components/VScrollButton.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/frontend/src/components/VScrollButton.vue b/frontend/src/components/VScrollButton.vue
--- a/frontend/src/components/VScrollButton.vue	(revision 431ccc43aaa8f58cacf18170c8f5cbddde48cbbd)
+++ b/frontend/src/components/VScrollButton.vue	(date 1727116542815)
@@ -24,45 +24,44 @@
 
 defineEmits<{ tab: [KeyboardEvent] }>()
 
-const ANALYTICS_ROUTES = [
-  "search__image",
-  "search__audio",
-  "search",
-  "image__collection",
-  "audio_collection",
-]
+const SEARCH_ROUTES = ["search-image", "search-audio", "search"]
+const ANALYTICS_ROUTES = [...SEARCH_ROUTES, "image-collection", "audio-collection"]
 
 const hClass = computed(() =>
   props.isFilterSidebarVisible ? positionWithSidebar : positionWithoutSidebar
 )
 const scrollToTop = () => {
+  const routeName = route.name?.toString().split("__")[0]
+
+  if (!routeName || !ANALYTICS_ROUTES.includes(routeName)) {
+    window.scrollTo({ top: 0, left: 0, behavior: "smooth" })
+    return
+  }
+
+  const isSearchRoute = routeName && SEARCH_ROUTES.includes(routeName)
+
   const mainPage = document.getElementById("main-page")
-  const element = mainPage || window
+
+  const scrollPixels = isSearchRoute ? mainPage?.scrollTop : window.scrollY
+  const maxScroll = isSearchRoute ? mainPage?.scrollHeight : document.body.scrollHeight
+
+  const element = isSearchRoute ? mainPage || window : window
   element.scrollTo({ top: 0, left: 0, behavior: "smooth" })
 
-  if (ANALYTICS_ROUTES.some((r) => route.name?.toString().startsWith(r))) {
     const kind: ResultKind =
       searchStore.strategy === "default" ? "search" : "collection"
     const collectionType = (searchStore.collectionValue as Collection) ?? "null"
 
     $sendCustomEvent("BACK_TO_TOP", {
-      searchType:
-        searchStore.searchType === "model-3d" ||
-        searchStore.searchType === "video"
-          ? "all"
-          : searchStore.searchType,
+      searchType: mediaStore._searchType,
       kind,
       query: searchStore.searchTerm,
       resultPage: mediaStore.currentPage,
-      scrollPixels: mainPage?.scrollTop || -1,
-      maxScroll:
-        mainPage && "scrollTopMax" in mainPage
-          ? (mainPage.scrollTopMax as number)
-          : -1,
+      scrollPixels: scrollPixels ?? -1,
+      maxScroll: maxScroll ?? -1,
       collectionType,
       collectionValue: searchStore.collectionValue ?? "null",
     })
-  }
 }
 </script>

```

**Review by obulat — APPROVED** [review]
I've added the suggested patch and also extracted the common properties for search-related analytics events.

Thank you for your contribution, @d3jawu !

**Pull request merged**